CenterForOpenScience/ember-osf-web
# [ENG-2554] Moderator editable metadata renderer

- Ticket: [ENG-2554](https://openscience.atlassian.net/browse/ENG-2554)
- Feature flag: n/a

## Purpose

Add the component that renders and edits the provider-specifc metadata.

### A registration with metadata and current user is moderator
<img width="1156" alt="Screen Shot 2021-03-19 at 3 13 01 PM" src="https://user-images.githubusercontent.com/6599111/111831429-bdd94600-88c5-11eb-85ef-9ad29f39f52e.png">

Click the edit button:
<img width="516" alt="Screen Shot 2021-03-19 at 3 16 40 PM" src="https://user-images.githubusercontent.com/6599111/111836758-2d9eff00-88cd-11eb-8ab1-17df1be2459a.png">

Make changes:
<img width="507" alt="Screen Shot 2021-03-19 at 3 16 48 PM" src="https://user-images.githubusercontent.com/6599111/111836773-3394e000-88cd-11eb-87c2-f96d63f0ffac.png">

Save changes:
<img width="279" alt="Screen Shot 2021-03-19 at 3 16 59 PM" src="https://user-images.githubusercontent.com/6599111/111836784-398ac100-88cd-11eb-8464-af062acbd027.png">


### A registration with [] metadata
<img width="1149" alt="Screen Shot 2021-03-19 at 1 03 17 PM" src="https://user-images.githubusercontent.com/6599111/111816775-9f6a4f00-88b3-11eb-9fe5-6e9d4f20ff2b.png">

### A registration with metadata and current user is not moderator
<img width="1147" alt="Screen Shot 2021-03-19 at 3 13 36 PM" src="https://user-images.githubusercontent.com/6599111/111831449-c5005400-88c5-11eb-992e-1fae1436b9b3.png">



## Summary of Changes

1. Add editable field for provider-metadata
2. Add provider metadata manager
3. Make editable and display versions of the provider metadata fields
4. Add provider metadata field into the registries metadata sidebar

To Do:
1. Tests

## Side Effects

Probably not. 

## QA Notes

This only appears on the registries overview page. States of interest are:

1. If there is no provider metadata specified on the registration provider, then this widget should not show at all.
2. If the user is not a moderator on the provider, then they should not have the edit icon.
3. If there is no metadata saved at all on the registration, but the provider does have metadata set, then it should show a message saying that there is no metadata.
4. If there is any metadata saved on the registration, then all fields should show, but the fields with no information will have a blank value.
